pkg/compactor/block_upload.go

```diff
@@ -365,6 +409,69 @@ func (c *MultitenantCompactor) uploadMeta(ctx context.Context, logger log.Logger
 	return nil
 }
 
+func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.ULID,
+	userBkt objstore.Bucket, meta metadata.Meta) error {
+
+	// create a temporary directory to download the block
+	// on normal compactors this will be in the compartor's data directory, but in other cases (like some tests)
+	// the fallback will be the system's temp directory
+	tmpDir := os.TempDir()
```

Comment:
[Nit] Isn't it simpler to just use `os.MkdirTemp`'s default for the parent directory?
```suggestion
	tmpDir := ""
```